### PR TITLE
Cleanup: Move Logger to base AuthenticationHandler + use more var

### PR DESCRIPTION
**Pull request by HaoK**
- Push Logger to base handler class/Initialize
- Pass to use var consistently

### CODE REVIEW

**Comment by HaoK** [comment]
@Tratcher

File: src/Microsoft.AspNet.Authentication/AuthenticationMiddleware.cs [hunk 1]
```diff
@@ -28,6 +30,8 @@ public abstract class AuthenticationMiddleware<TOptions> where TOptions : Authen
             {
                 Options = options.Options;
             }
+            Logger = loggerFactory.CreateLogger(this.GetType().FullName);
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Authentication/AuthenticationMiddleware.cs]: This gets the name of the derived type, correct?

**Comment by Tratcher** [comment]
Squash and :shipit:

File: src/Microsoft.AspNet.Authentication/AuthenticationMiddleware.cs (hunk 1 above)
Comment (HaoK) [inline src/Microsoft.AspNet.Authentication/AuthenticationMiddleware.cs]: Yah it gets the actual type that's being constructed.

**Comment by HaoK** [comment]
30d350da26d5c40e9e160ccb2cd120303da5baf1

**Pull request closed without merging**